### PR TITLE
Fixed: Validate if an RFExtractionAI can be used for extraction rather than crashing

### PR DESCRIPTION
**Pull request by samuel500**
This creates a method `check_is_ready_for_extraction` to validate if an RFExtractionAI can be used for extraction, and fixes a few minor issues:
- The `_meta_data` attribute is now also removed when running `Project.lose_weight`
- Saving an ExtractionAI object with `reduce_weight` set to True and `keep_documents` set to False does not remove the documents in memory. 
- The `category` can now be set in the parent `Trainer` class.
- Removes a few legacy methods.
- Removes the `pathos` dependency.

### CODE REVIEW

File: konfuzio_sdk/trainer/information_extraction.py [hunk 1]
```diff
@@ -1207,93 +1207,33 @@ def __init__(self, *args, **kwargs):
         """Initialize ExtractionModel."""
         # Go through keyword arguments, and either save their values to our
         # instance, or raise an error.
+        self.category: Category = kwargs.get('category', None)
+        logger.info(f"{self.category=}")
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/information_extraction.py:1211]: What is the strategy of this logger statement? Seems like they are only temporary for debugging. If they stay they need to be a full sentence

File: konfuzio_sdk/trainer/information_extraction.py [hunk 2]
```diff
@@ -1207,93 +1207,33 @@ def __init__(self, *args, **kwargs):
         """Initialize ExtractionModel."""
         # Go through keyword arguments, and either save their values to our
         # instance, or raise an error.
+        self.category: Category = kwargs.get('category', None)
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/information_extraction.py:1210]: What is the backgroud to adding 'category'?
Why do we need it to parse from kwargs, should we make it a named argument?
Is it usually expected?

File: konfuzio_sdk/data.py [hunk 3]
```diff
@@ -2879,6 +2880,7 @@ def lose_weight(self):
         for label in self.labels:
             label.lose_weight()
         self._documents = []
+        self._meta_data = []
```
Comment (zypriafl) [inline konfuzio_sdk/data.py:2883]: What is this use for?

Comment (samuel500) [reply konfuzio_sdk/data.py:2883]: It's where the list of all the documents is loaded into from the Project meta-data file (`documents_meta.json5`).

File: konfuzio_sdk/trainer/information_extraction.py (hunk 1 above)
Comment (samuel500) [reply konfuzio_sdk/trainer/information_extraction.py:1211]: Is there a good template for when one is logging multiple arguments at the beginning of a method or when initialising a class? For example, when I initialise RFExtractionAI, https://github.com/konfuzio-ai/konfuzio-sdk/blob/master/konfuzio_sdk/trainer/information_extraction.py#L1951
would just adding a log at the beginning like: `logger.info("Initializing RFExtractionAI with settings:")` be enough? Or should I write a sentence for each?

**Pull request merged**